File: Java/api/org/rr0/im/business/report/Account.java
package org.rr0.im.business.report;

import org.rr0.im.business.event.Timeable;

/**
 * A set of reported Events by a Source.
 *
 * @author Jérôme Beau
 * @version 21 avr. 2003 15:56:42
 */
public interface Account extends Timeable {
    /**
     * The reporting source. May be some people, a newspaper article, etc.
     *
     * @associates <{org.rr0.im.business.report.Source}>
     * @supplierRole reportsource
     */
    Source getSource();

    void setSource (Source source);

    /**
     * Get the accounts contents
     *
     * @return The text of the free report.
     */
    Object getContent();

    void setContent(Object content);
}

File: impl/src/org/rr0/im/business/report/Account.ts
import {Source} from "../../../../../../../api/src/org/rr0/im/business/report/Source";
import {Moment} from "../../../../../../../api/src/org/rr0/im/business/event/circumstance/Moment";
import {Event} from "../event/Event";

/**
 * Account Reference Implementation.
 */
export abstract class Account extends Event {
  /**
   * The reporting source. May be some people, a newspaper article, etc.
   */
  source: Source;

  /**
   * The account contents
   */
  content: Object;

  constructor(date: Moment, source: Source) {
    super(date);
    this.source = source;
  }
}
